workers/data_refinery_workers/processors/qn_reference.py

```diff
@@ -35,24 +35,68 @@ def _prepare_input(job_context: Dict) -> Dict:
     # I'm not crazy about this yet. Maybe refactor later,
     # but I need the data now.
     job_context = smasher._prepare_files(job_context)
-    job_context = smasher._smash(job_context)
-
-    if not 'final_frame' in job_context.keys():
-        logger.error("Unable to prepare files for creating QN target.",
-            job_id=job_context['job'].id)
-        job_context["job"].failure_reason = "Couldn't prepare files creating QN target (no final_frame)."
-        job_context['success'] = False
-        return job_context
 
     # work_dir is already created by smasher._prepare_files
     outfile_base = job_context['work_dir'] + str(time.time()).split('.')[0]
-    outfile = outfile_base + '.tsv'
-    job_context['final_frame'].to_csv(outfile, sep='\t', encoding='utf-8')
-    job_context['smashed_file'] = outfile
     job_context['target_file'] = outfile_base + '_target.tsv'
 
     return job_context
 
+def _build_qn_target(job_context: Dict) -> Dict:
+    """ Iteratively creates a QN target file, method described here: https://github.com/AlexsLemonade/refinebio/pull/1013"""
+
+    job_context['time_start'] = timezone.now()
+
+    # Get the gene list from the first input
+    geneset_target_frame = pd.read_csv(job_context['input_files']['ALL'][0].get_synced_file_path(), sep='\t', header=0,
+                                          index_col=0, error_bad_lines=False)
+
+    # Get the geneset
+    geneset = geneset_target_frame.index.values
+
+    # Sequentially build the target
+    sum_frame_input = {'index': geneset, 'sum': [0 for value in geneset]}
+    sum_frame = pd.DataFrame(data=sum_frame_input)
+    sum_frame = sum_frame.set_index('index')
+
+    # Read and sum all of the inputs
+    num_valid_inputs = 0
+    for file in job_context['input_files']['ALL']:
+        input_frame = pd.read_csv(file.get_synced_file_path(), sep='\t', header=0,
+                                          index_col=0, error_bad_lines=False)
+
+        # If this input doesn't have the same geneset, we don't want it!
+        if list(input_frame.index.values) != list(geneset):
+            logger.error("Input frame doesn't match target geneset, skipping!",
+                bad_file=file,
+                target_geneset=geneset,
+                bad_geneset=input_frame.index.values)
```

Comment:
These should be a huge red flag. If we have these, our SCAN processor isn't producing what we expect in all cases (could happen if CEL files != what's annotated and we trust what's annotated). These will be very important to track down.